routing/chainview/bitcoind.go

```diff
@@ -0,0 +1,458 @@
+package chainview
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/roasbeef/btcd/btcjson"
+	"github.com/roasbeef/btcd/chaincfg"
+	"github.com/roasbeef/btcd/chaincfg/chainhash"
+	"github.com/roasbeef/btcd/rpcclient"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/roasbeef/btcutil"
+	"github.com/roasbeef/btcwallet/chain"
+	"github.com/roasbeef/btcwallet/wtxmgr"
+)
+
+// BitcoindFilteredChainView is an implementation of the FilteredChainView
+// interface which is backed by bitcoind.
+type BitcoindFilteredChainView struct {
+	started int32
+	stopped int32
+
+	// bestHeight is the height of the latest block added to the
+	// blockQueue from the onFilteredConnectedMethod. It is used to
+	// determine up to what height we would need to rescan in case
+	// of a filter update.
+	bestHeightMtx sync.Mutex
+	bestHeight    uint32
+
+	chainClient *chain.BitcoindClient
```

Comment:
Most of this is copied verbatim from the implementation for `btcd` It seems that we can re-use most of the code by creating an interface in place of `chainClient`. So we would have something like a `NodeFilteredView`, and then we'd create a `BitcoinFilteredChainView` and `BtcdFilteredChainView` by creation an instance of `NodeFilteredView` with the proper concrete implementation of the interface.

Reply:
I think we can just add this as a TODO now, then follow back on this once we've merged in this PR.

Reply:
I've added this as a TODO.